#This script details the URL and references of publicly available datasets and softwares used in this project.

##Datasets used
######################################
1.Single-cell RNA-seq dataset from Tucker et. al.
URL: https://www.ahajournals.org/doi/10.1161/CIRCULATIONAHA.119.045401

2.Bulk RNA-seq dataset from Tan et. al.
URL: https://www.ahajournals.org/doi/10.1161/CIRCRESAHA.120.317254

##Software used:
######################################
1.Seurat 3.1.5 used for guided clustering
URL: https://satijalab.org/seurat/

2.DAVID 6.8 used for pathway analysis
URL: https://david.ncifcrf.gov/

3.BisqueRNA 1.0.3 used for bulk tissue deconvolution
URL: https://cran.r-project.org/web/packages/BisqueRNA/vignettes/bisque.html

4.R 3.6.3 used for codes detailed in this project
URL: https://www.r-project.org/

##References:
######################################
- Tucker, N. R., et al. (2020). "Transcriptional and Cellular Diversity of the Human Heart." Circulation.
- Tan, W. L. W., et al. (2020). "Epigenomes of Human Hearts Reveal New Genetic Variants Relevant for Cardiac Disease and Phenotype." Circ Res.
- Butler, A., et al. (2018). "Integrating single-cell transcriptomic data across different conditions, technologies, and species." Nat Biotechnol 36(5): 411-420.
- Huang, D. W., et al. (2007). "The DAVID Gene Functional Classification Tool: a novel biological module-centric algorithm to functionally analyze large gene lists." Genome Biol 8(9): R183.
- Jew, B., et al. (2020). "Accurate estimation of cell composition in bulk expression through robust integration of single-cell information." Nat Commun 11(1): 1971.



