#This script details the URL and references of publicly available datasets and softwares used in this project.

##Datasets used
######################################
1.Single-cell RNA-seq dataset from Tucker et. al.
URL: https://www.ahajournals.org/doi/10.1161/CIRCULATIONAHA.119.045401

2.Bulk RNA-seq dataset from Tan et. al.
URL: https://www.ahajournals.org/doi/10.1161/CIRCRESAHA.120.317254

##Software used:
######################################
1.Seurat 3.1.5 used for guided clustering
URL: https://satijalab.org/seurat/

2.DAVID 6.8 used for pathway analysis
URL: https://david.ncifcrf.gov/

3.BisqueRNA 1.0.3 used for bulk tissue deconvolution
URL: https://cran.r-project.org/web/packages/BisqueRNA/vignettes/bisque.html

4.R 3.6.3 used for codes detailed in this project
URL: https://www.r-project.org/

##References:
######################################
Tucker, N. R., et al. (2020). "Transcriptional and Cellular Diversity of the Human Heart." Circulation.
	Background: The human heart requires a complex ensemble of specialized cell types to perform its essential function. A greater knowledge of the intricate cellular milieu of the heart is critical to increase our understanding of cardiac homeostasis and pathology. As recent advances in low input RNA-sequencing have allowed definitions of cellular transcriptomes at single cell resolution at scale, here we have applied these approaches to assess the cellular and transcriptional diversity of the non-failing human heart. Methods: Microfluidic encapsulation and barcoding was used to perform single nuclear RNA sequencing with samples from seven human donors, selected for their absence of overt cardiac disease. Individual nuclear transcriptomes were then clustered based upon transcriptional profiles of highly variable genes. These clusters were used as the basis for between-chamber and between-sex differential gene expression analyses and intersection with genetic and pharmacologic data. Results: We sequenced the transcriptomes of 287,269 single cardiac nuclei, revealing a total of 9 major cell types and 20 subclusters of cell types within the human heart. Cellular subclasses include two distinct groups of resident macrophages, four endothelial subtypes, and two fibroblasts subsets. Comparisons of cellular transcriptomes by cardiac chamber or sex reveal diversity not only in cardiomyocyte transcriptional programs, but also in subtypes involved in extracellular matrix remodeling and vascularization. Using genetic association data, we identified strong enrichment for the role of cell subtypes in cardiac traits and diseases. Finally, intersection of our dataset with genes on cardiac clinical testing panels and the druggable genome reveals striking patterns of cellular specificity. Conclusions: Using large-scale single nuclei RNA sequencing, we have defined the transcriptional and cellular diversity in the normal human heart. Our identification of discrete cell subtypes and differentially expressed genes within the heart will ultimately facilitate the development of new therapeutics for cardiovascular diseases.
Tan, W. L. W., et al. (2020). "Epigenomes of Human Hearts Reveal New Genetic Variants Relevant for Cardiac Disease and Phenotype." Circ Res.
	Rationale: Identifying genetic markers for heterogeneous complex diseases such as heart failure is challenging, and requires prohibitively large cohort sizes in genome-wide association studies (GWAS) in order to meet the stringent threshold of genome-wide statistical significance. On the other hand, chromatin quantitative trait loci (QTL), elucidated by direct epigenetic profiling of specific human tissues, may contribute towards prioritising sub-threshold variants for disease-association. Objective: Here, we captured non-coding genetic variants by performing epigenetic profiling for enhancer H3K27ac ChIP-seq in 70 human control and end-stage failing hearts. Methods and Results: We have mapped a comprehensive catalogue of 47,321 putative human heart enhancers and promoters. 3,897 differential acetylation peaks (FDR 5%) pointed to pathways altered in heart failure (HF). To identify cardiac histone acetylation QTLs (haQTLs), we regressed out confounding factors including HF disease status, and employed the G-SCI test1 to call out 1,680 haQTLs (FDR 10%). RNA-seq performed on the same heart samples proved a subset of haQTLs to have significant association also to gene expression (expression QTLs), either in cis (180), or through long range interactions (81), identified by Hi-C and HiChIP performed on a subset of hearts. Furthermore, a concordant relationship between the gain or disruption of transcription factor (TF) binding motifs, inferred from alternative alleles at the haQTLs, implied a surprising direct association between these specific TF and local histone acetylation in human hearts. Finally, 62 unique loci were identified by colocalisation of haQTLs with the sub-threshold loci of heart-related GWAS datasets. Conclusions: Disease and phenotype-association for 62 unique loci are now implicated. These loci may indeed mediated their effect through modification of enhancer H3K27-acetylation enrichment and their corresponding gene expression differences.
Butler, A., et al. (2018). "Integrating single-cell transcriptomic data across different conditions, technologies, and species." Nat Biotechnol 36(5): 411-420.
	Computational single-cell RNA-seq (scRNA-seq) methods have been successfully applied to experiments representing a single condition, technology, or species to discover and define cellular phenotypes. However, identifying subpopulations of cells that are present across multiple data sets remains challenging. Here, we introduce an analytical strategy for integrating scRNA-seq data sets based on common sources of variation, enabling the identification of shared populations across data sets and downstream comparative analysis. We apply this approach, implemented in our R toolkit Seurat (http://satijalab.org/seurat/), to align scRNA-seq data sets of peripheral blood mononuclear cells under resting and stimulated conditions, hematopoietic progenitors sequenced using two profiling technologies, and pancreatic cell 'atlases' generated from human and mouse islets. In each case, we learn distinct or transitional cell states jointly across data sets, while boosting statistical power through integrated analysis. Our approach facilitates general comparisons of scRNA-seq data sets, potentially deepening our understanding of how distinct cell states respond to perturbation, disease, and evolution.
Huang, D. W., et al. (2007). "The DAVID Gene Functional Classification Tool: a novel biological module-centric algorithm to functionally analyze large gene lists." Genome Biol 8(9): R183.
	The DAVID Gene Functional Classification Tool http://david.abcc.ncifcrf.gov uses a novel agglomeration algorithm to condense a list of genes or associated biological terms into organized classes of related genes or biology, called biological modules. This organization is accomplished by mining the complex biological co-occurrences found in multiple sources of functional annotation. It is a powerful method to group functionally related genes and terms into a manageable number of biological modules for efficient interpretation of gene lists in a network context.
Jew, B., et al. (2020). "Accurate estimation of cell composition in bulk expression through robust integration of single-cell information." Nat Commun 11(1): 1971.
	We present Bisque, a tool for estimating cell type proportions in bulk expression. Bisque implements a regression-based approach that utilizes single-cell RNA-seq (scRNA-seq) or single-nucleus RNA-seq (snRNA-seq) data to generate a reference expression profile and learn gene-specific bulk expression transformations to robustly decompose RNA-seq data. These transformations significantly improve decomposition performance compared to existing methods when there is significant technical variation in the generation of the reference profile and observed bulk expression. Importantly, compared to existing methods, our approach is extremely efficient, making it suitable for the analysis of large genomic datasets that are becoming ubiquitous. When applied to subcutaneous adipose and dorsolateral prefrontal cortex expression datasets with both bulk RNA-seq and snRNA-seq data, Bisque replicates previously reported associations between cell type proportions and measured phenotypes across abundant and rare cell types. We further propose an additional mode of operation that merely requires a set of known marker genes.






